File: src/main/java/tafqeetj/numbers/positive/PositiveBasicNumbersPLacesNames.java
package tafqeetj.numbers.positive;

import tafqeetj.numbers.BasicNumberPlacesNames;

import java.util.Map;

public class PositiveBasicNumbersPLacesNames extends BasicNumberPlacesNames {

    private static PositiveBasicNumbersPLacesNames instance;
    private PositiveBasicNumbersPLacesNames() {}

    public synchronized static PositiveBasicNumbersPLacesNames getInstance() {
        if (instance == null)
            instance = new PositiveBasicNumbersPLacesNames();
        return instance;
    }
}

File: src/main/kotlin/tech/moussa/tafqeetk/numbers/positive/PositiveBasicNumbersPlacesNames.kt
package tech.moussa.tafqeetk.numbers.positive

import tech.moussa.tafqeetk.numbers.BasicNumberPlacesNames

class PositiveBasicNumbersPlacesNames private constructor(): BasicNumberPlacesNames() {
    companion object {
        private var instance: PositiveBasicNumbersPlacesNames? = null

        @Synchronized
        fun getInstance(): PositiveBasicNumbersPlacesNames {
            if (PositiveBasicNumbersPlacesNames.instance == null) {
                PositiveBasicNumbersPlacesNames.instance =
                        PositiveBasicNumbersPlacesNames()
            }
            return PositiveBasicNumbersPlacesNames.instance!!
        }
    }
}